File: ABC/351-400/368/b.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i,n) for(int i=0;i<n;++i)

int main(){
    int n; cin >> n;
    vector<int> a(n);
    rep(i,n) cin >> a[i];
    int ans = 0;
    while(true){
        ans++;
        sort(a.rbegin(),a.rend());
        a[0]--; a[1]--;
        int cnt = 0;
        rep(i,n) if(a[i] > 0) cnt++;
        if(cnt <= 1) break;
    }
    cout << ans << endl;
    return 0;
}

File: ABC/351-400/368/b.py
n = int(input())
a = list(map(int,input().split()))
ans = 0

while True:
    a.sort(reverse=True)
    a[0]-=1
    a[1]-=1
    cnt = 0
    ans+=1
    for x in a:
        if x > 0:
            cnt+=1
    
    if cnt <= 1:
        break
    
print(ans)

